### PR TITLE
Create closure expiry event generation

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/Processor.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.nationalarchives
+
+import cats.effect.IO
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse
+import uk.gov.nationalarchives.Lambda.Config
+import uk.gov.nationalarchives.dp.client.ContentClient
+import io.circe.generic.auto._
+import cats.implicits._
+
+object Processor {
```
Comment (techncl) [inline src/main/scala/uk/gov/nationalarchives/Processor.scala]: Do you think that maybe we can rename this to something more specific like `SqsSender`, `SqsMessageSender`?

File: src/main/scala/uk/gov/nationalarchives/Processor.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def processDocuments(
+      sqsClient: DASQSClient[IO],
+      contentClient: ContentClient[IO],
+      config: Config
+  ): IO[List[SendMessageResponse]] = for {
+    documents <- contentClient.findExpiredClosedDocuments(config.secretName)
```
Comment (techncl) [inline src/main/scala/uk/gov/nationalarchives/Processor.scala]: Maybe "documents" could be renamed to `expiredEntities` or `expiredClosedEntities`

File: src/main/scala/uk/gov/nationalarchives/Processor.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import cats.effect.IO
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse
+import uk.gov.nationalarchives.Lambda.Config
+import uk.gov.nationalarchives.dp.client.ContentClient
+import io.circe.generic.auto._
+import cats.implicits._
+
+object Processor {
+  def processDocuments(
```
Comment (techncl) [inline src/main/scala/uk/gov/nationalarchives/Processor.scala]: Maybe `sendEntities` or `sendExpiredEntities` instead of `processDocuments`?

**Review by techncl — COMMENTED** [review]
Looks good; just a few comments about naming.

File: src/main/scala/uk/gov/nationalarchives/Processor.scala (hunk 1 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: It's not only sending messages though, it's getting the entities from Preservica as well and it gets a bit verbose describing both things which is why I kept it generic. I'm happy to change it but it'd need to reflect the fact that it gets entities from Preservica.

File: src/main/scala/uk/gov/nationalarchives/Processor.scala (hunk 2 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: `expiredClosedEntities` works. I've changed it.

File: src/main/scala/uk/gov/nationalarchives/Processor.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: As with the name of the object, it's not just sending entities, it's fetching them as well. I can change it but I'm not sure what to.

Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: hmm, (a bit verbose but) `getExpiredEntitiesAndSendToSqs` or `findExpiredEntitiesAndSendToSqs`?

File: src/main/scala/uk/gov/nationalarchives/Processor.scala (hunk 1 above)
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: I see, maybe `EntitiesProcessor`, `ExpiredEntitiesProcessor`, `EntitiesHandler` or `ExpiredEntitiesHandler`

Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: `ExpiredEntitiesProcessor` is good, I've changed it.

File: src/main/scala/uk/gov/nationalarchives/Processor.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/Processor.scala]: I've used `findExpiredEntitiesAndSendToSqs` `get...` sounds very Java-y :)

**Review by techncl — APPROVED** [review]
Looks good! Thanks.

**Review by techncl — APPROVED** [review]

**Review by techncl — APPROVED** [review]

**Pull request merged**